lib/Sema/CSGen.cpp

```diff
@@ -1286,6 +1257,19 @@ namespace {
       return witnessType;
     }
 
+    Type visitRegexLiteralExpr(RegexLiteralExpr *E) {
+      auto &ctx = CS.getASTContext();
+      auto *regexDecl = ctx.getRegexDecl();
+      auto substringType = ctx.getSubstringType();
+      auto dynCapturesType = ctx.getDynamicCapturesType();
+      // TODO: Replace `(Substring, DynamicCaptures)` with type inferred from
+      // the regex.
+      auto matchType = TupleType::get({substringType, dynCapturesType}, ctx);
+      auto type = BoundGenericStructType::get(regexDecl, Type(), {matchType});
+      CS.setType(E, type);
```

Comment:
Nit: You don't have to explicitly set the type, as the caller will do that

Reply:
Thanks. Fixed.